Userland/Libraries/LibCore/ConfigFile.cpp

```diff
@@ -18,48 +18,44 @@ ErrorOr<NonnullRefPtr<ConfigFile>> ConfigFile::open_for_lib(String const& lib_na
 {
     String directory = StandardPaths::config_directory();
     auto path = String::formatted("{}/lib/{}.ini", directory, lib_name);
-
-    return adopt_nonnull_ref_or_enomem(new (nothrow) ConfigFile(path, allow_altering));
+    return ConfigFile::open(path, allow_altering);
 }
 
 ErrorOr<NonnullRefPtr<ConfigFile>> ConfigFile::open_for_app(String const& app_name, AllowWriting allow_altering)
 {
     String directory = StandardPaths::config_directory();
     auto path = String::formatted("{}/{}.ini", directory, app_name);
-    return adopt_nonnull_ref_or_enomem(new (nothrow) ConfigFile(path, allow_altering));
+    return ConfigFile::open(path, allow_altering);
 }
 
 ErrorOr<NonnullRefPtr<ConfigFile>> ConfigFile::open_for_system(String const& app_name, AllowWriting allow_altering)
 {
     auto path = String::formatted("/etc/{}.ini", app_name);
-    return adopt_nonnull_ref_or_enomem(new (nothrow) ConfigFile(path, allow_altering));
+    return ConfigFile::open(path, allow_altering);
 }
 
 ErrorOr<NonnullRefPtr<ConfigFile>> ConfigFile::open(String const& filename, AllowWriting allow_altering)
 {
-    return adopt_nonnull_ref_or_enomem(new (nothrow) ConfigFile(filename, allow_altering));
+    auto file = File::construct(filename);
+    if (!file->open(allow_altering == AllowWriting::Yes ? OpenMode::ReadWrite : OpenMode::ReadOnly)) {
+        // Failure to open a read-only file is OK, and behaves as if the file was empty.
+        if (allow_altering == AllowWriting::Yes)
+            return Error::from_string_literal("Unable to open config file");
```

Comment:
Is this a new behavior?

Reply:
No. Before the constructor would create a File object, and then attempt to `open()` it but ignore failure. So this is the same effect done a different way. (Not ignoring the failure was causing a lot of crashes too - several places rely on this behavior.)